script.js

```diff
@@ -1,12 +1,206 @@
+    const inputValue = event.target.value;
+    rootElem.textContent = "";
+
+    const filteredEpisodes = episodes.filter((episode) => {
+      return (
+        episode.summary.toLowerCase().includes(inputValue.toLowerCase()) ||
+        episode.name.toLowerCase().includes(inputValue.toLowerCase())
+      );
+    });
+    makePageForEpisodes(filteredEpisodes);
+    numberOfEpisodes.innerText = `found ${filteredEpisodes.length} episodes`;
+  });
+}
+// level 300
+function addOptions(episodes) {
+  selectElement.innerHTML = "";
+  episodes.forEach((episode) => {
+    const optionElement = document.createElement("option");
+    optionElement.innerHTML = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")} - ${
+      episode.name
+    }`;
+    selectElement.appendChild(optionElement);
+  });
+}
+function buttonBackToAllEpisodes(episodes) {
+  const backButton = document.createElement("button");
+  backButton.classList.add("back-button");
+  backButton.innerText = "Show all episodes";
+  rootElem.appendChild(backButton);
+  backButton.addEventListener("click", (event) => {
+    rootElem.textContent = "";
+    makePageForEpisodes(episodes);
+    backButton.remove();
+  });
+}
+//level 350
+
+function fetchEpisodes(episodeNumber) {
+  fetch(`https://api.tvmaze.com/shows/${episodeNumber}/episodes`)
+    .then((response) => response.json())
+    .then((data) => {
+      makePageForEpisodes(data);
+    })
+    .catch((error) => console.log(error));
+}
+//Level 400
+function addOptionsShow() {
+  let showTitles = shows.map((show) => show.name).sort();
+  showTitles.forEach((show) => {
+    const optionElement = document.createElement("option");
+    optionElement.innerHTML = show;
+    selectElementShow.appendChild(optionElement);
+  });
+}
+
+function chooseAnEpisode(episodes) {
+  selectElement.addEventListener("change", (event) => {
+    const inputValue = event.target.value;
+    rootElem.textContent = "";
+    const filteredEpisodes = episodes.filter((episode) =>
+      inputValue.includes(episode.name)
+    );
+    makePageForEpisodes(filteredEpisodes);
+    buttonBackToAllEpisodes(episodes);
+  });
+}
+function chooseShow(classOrId, eventListener) {
+  classOrId.addEventListener(eventListener, (event) => {
+    const inputValue = event.target.value;
+
+    rootElem.textContent = "";
+    rootElemShow.textContent = "";
+    let filteredShow = shows.filter((show) => show.name === inputValue);
+
+    fetch(`https://api.tvmaze.com/shows/${filteredShow[0].id}/episodes`)
+      .then((response) => response.json())
+      .then((data) => {
+        makePageForEpisodes(data);
+      })
+      .catch((error) => console.log(error));
+  });
 }
 
+//Level 500
+function showAllShows(shows) {
+  selectElementShow.style.display = "inline";
+  selectElement.style.display = "none";
+  shows.forEach((show) => {
+    const showContainer = document.createElement("div");
```

Comment:
There are a lot of calls here that do the same thing but slightly differently, which are a bit hard to follow; lots of calls to `document.createElement` then `element.classList =` then `element.innerText =` then `parent.appendChild(element)` - can you think of a useful function to extract to make it more clear what's happening each time?